### DISCUSSION TITLE
Apply MONAI transforms on costume data

### DISCUSSION THREAD

**Discussion by cugwu** [Project-MONAI/tutorials#1592] [Q&A] [answered]
Hi to everyone, I have a dataset of CT scans saved as a .tiff file. The data are stored in a particular way needing a customised Dataset class which is an extension of torch.utils.data.Dataset. The __getitem__ function of my class loads the data and applies the required transformations. When I load the data I have a torch.tensor of size x,y,z (meaning no channel dimensions). The transforms that I apply are:
```
        train_transforms = Compose(
            [
                EnsureChannelFirst,
                Spacing(pixdim=(args.space_x, args.space_y, args.space_z), mode=("bilinear", "nearest")),
                ScaleIntensityRange(a_min=args.a_min, a_max=args.a_max, b_min=args.b_min, b_max=args.b_max, clip=True),
                SpatialPad(spatial_size=[args.roi_x, args.roi_y, args.roi_z]),
                RandSpatialCropSamples(
                    roi_size=[args.roi_x, args.roi_y, args.roi_z],
                    num_samples=args.sw_batch_size,
                    random_center=True,
                    random_size=False,
                ),
            ]
```
But I get the error: ValueError: Metadata not available and channel_dim=None, EnsureChannelFirst is not in use.
What does my tensor need to have to be able to use monai.ransforms without any problem? What metadata  EnsureChannelFirst 
 is expecting? Am I doing something wrong? There shouldn't be problems applying the transform to a tensor since MONAI is based on pytorch. Do I have to extend particular Dataset classes from MONAI?
Thanks for the answer!

**Comment by KumoLiu** [comment, marked as answer]
Hi @cugwu, thanks for your interest here.
You can specify the `channel_dim` in the `EnsureChannelFirst`.
```
EnsureChannelFirst(channel_dim="no_channel")
```

Hope it helps, thanks!
